### PR TITLE
test(NODE-4282): sync fle2 payload spec test updates

### PR DESCRIPTION
**Pull request by durran**
### Description

Fixes FLE2 CRUD tests with payload updates.

#### What is changing?

Syncs the latest tests and pins the checkout of libmongocrypt to 4c3715bbe360a6e87f411d63f5bd1dc237b2f5cb (1.5.0)

##### Is there new documentation needed for these changes?

None

#### What is the motivation for this change?

NODE-4282

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the correct format: `<type>(NODE-xxxx)<!>: <description>`
- [x] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

**Review by dariakp — COMMENTED** [review]
Just have a question here

File: .evergreen/run-custom-csfle-tests.sh [hunk 1]
```diff
@@ -27,7 +27,7 @@ ABS_PATH_TO_PATCH=$(pwd)
 # CSFLE_GIT_REF - set the git reference to checkout for a custom CSFLE version
 # CDRIVER_GIT_REF - set the git reference to checkout for a custom CDRIVER version (this is for libbson)
 
-CSFLE_GIT_REF=${CSFLE_GIT_REF:-master}
+CSFLE_GIT_REF=${CSFLE_GIT_REF:-4c3715bbe360a6e87f411d63f5bd1dc237b2f5cb}
```
[the remark this answers is not in the document]
Comment (durran) [reply .evergreen/run-custom-csfle-tests.sh:30]: 7 characters are the default for a git short SHA, but it was easier for me to copy the entire SHA from the git log than select the first 7 so I used the entire thing.

**Comment by durran** [comment]
These errors: `Uncaught TypeError: contention factor is required for indexed algorithm` need to be fixed in libmongocrypt in another ticket.

**Review by dariakp — DISMISSED** [review]

**Comment by durran** [comment]
Once the alpha.4 artefacts are published to Github I'll update this PR to point at that and the regular fle tests will pass with the exception of the 1 timeout failure. Looking into that now.

**Review by nbbeeken — APPROVED** [review]

**Pull request merged**